### PR TITLE
test: Split rpc_signmessage test for disabled wallet

### PR DESCRIPTION
**Pull request by Shubhankar-Gambhir**
This PR enables a part of the non-wallet functional test  (rpc_signmessage.py) to be run even with the Bitcoin Core wallet disabled, it is inspired by #20078.

Divided tests in rpc_signmessage.py into 2 files wallet_signmessagewithaddress.py and rpc_signmessagewithprivkey.py, latter one can run even when wallet is disabled that provides extra test which was not performed earlier.

* we need bitcoincore wallet to run  rpc_signmessage.py, but it is olny required for signing messages with address and not for signing messages wih private key, so latter one can be in a seperate test which can run without wallet
* verifying message doesn't require wallet, so it can be used in both tests without any problem
* 2 tests are named as wallet_signmessagewithaddress.py and rpc_signmessagewithprivkey.py to provide clarity of what they are testing.

### CODE REVIEW

File: test/functional/test_runner.py [hunk 1]
```diff
@@ -121,7 +121,6 @@
     'wallet_listreceivedby.py --legacy-wallet',
     'wallet_listreceivedby.py --descriptors',
     'wallet_abandonconflict.py --legacy-wallet',
-    'p2p_dns_seeds.py',
```
Comment (Shubhankar-Gambhir) [inline test/functional/test_runner.py:124]: including p2p_dns_seed.py

File: test/functional/test_runner.py [hunk 2]
```diff
@@ -224,7 +224,8 @@
     'wallet_importprunedfunds.py --descriptors',
     'p2p_leak_tx.py',
     'p2p_eviction.py',
-    'rpc_signmessage.py',
+    'rpc_signmessagewithaddress.py',
```
Comment (MarcoFalke) [inline test/functional/test_runner.py:227]: wallet tests should to be prefixed with `wallet_` (see our docs on prefixes)

**Review by maflcko — COMMENTED** [review]
Please squash your commits according to https://github.com/bitcoin/bitcoin/blob/master/CONTRIBUTING.md#squashing-commits

File: test/functional/test_runner.py (hunk 2 above)
Comment (Shubhankar-Gambhir) [reply test/functional/test_runner.py]: Done, thanks for pointing out.

File: test/functional/rpc_signmessagewithprivkey.py [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+        priv_key = 'cUeKHd5orzT3mz8P9pxyREHfsWtVfgsfDjiZZBcjUBAaGk1BTj7N'
+        address = 'mpLQjfK79b7CCV4VMJWEWAj5Mpx8Up5zxB'
+        expected_signature = 'INbVnW4e6PeRmsv2Qgu8NuopvrVjkcxob+sX8OcZG0SALhWybUjzMLPdAsXI46YZGb0KQTRii+wWIQzRpG/U+S0='
+        signature = self.nodes[0].signmessagewithprivkey(priv_key, message)
+        assert_equal(expected_signature, signature)
+        assert self.nodes[0].verifymessage(address, signature, message)
+
+        self.log.info('test parameter validity and error codes')
+        # signmessage(withprivkey) have two required parameters
+        for num_params in [0, 1, 3, 4, 5]:
```
Comment (ShubhamPalriwala) [inline test/functional/rpc_signmessagewithprivkey.py:32]: Is there a reason why we're running this from 0 to 5?

Comment (ShubhamPalriwala) [inline test/functional/rpc_signmessagewithprivkey.py:32]: Is there a reason for running the loop here on L32 and L36 from 0 to 5?

**Review by ShubhamPalriwala — COMMENTED** [review]
tAck 17019494d9ae381fa75bf469ca10bdad95df7ae1

Decoupling the signing of messages with a private key and with an address opens up the doors for further individual testing in the future.

Tested on Ubuntu 21.04

File: test/functional/rpc_signmessagewithprivkey.py [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+from test_framework.util import (
+    assert_equal,
+    assert_raises_rpc_error,
+)
+
+class SignMessagesWithPrivTest(BitcoinTestFramework):
+    def set_test_params(self):
+        self.setup_clean_chain = True
+        self.num_nodes = 1
+        self.extra_args = [["-addresstype=legacy"]]
```
Comment (laanwj) [inline test/functional/rpc_signmessagewithprivkey.py:17]: I don't think this argument makes any difference without wallet?

**Review by vasild — COMMENTED** [review]
Approach ACK

I verified that this is just moving bits of a test to a separate new test (as intended).

Tested:

When compiled with `--enable-wallet`:
```
rpc_signmessagewithprivkey.py    | ✓ Passed  | 1 s
wallet_signmessagewithaddress.py | ✓ Passed  | 1 s
```

When compiled with `--disable-wallet`:
```
rpc_signmessagewithprivkey.py    | ✓ Passed  | 1 s
wallet_signmessagewithaddress.py | ○ Skipped | 0 s
```

`wallet_signmessagewithaddress.py` fails as expected if `skip_if_no_wallet()` is removed from it and the wallet is not enabled.

Just a few minor suggestions:

* The first line in the commit message usually starts with a lowercase letter and has no space before the `:`:
```sh
$ git log --oneline --grep='^test:' origin/master |wc -l
     829

$ git log --oneline --grep='^Test:' origin/master |wc -l
       3

$ git log --oneline --grep='^Test :' origin/master |wc -l
       0
```

* In the OP: `s/rpc_signmessagewithaddress.py/wallet_signmessagewithaddress.py/`

* The commit message is a bit scarce. In general, it is good if it contains a summary of the changes and more importantly, _why_ are they being done. It would be best if one can make sense of it by just reading the commit message, if github.com is not available.

File: test/functional/rpc_signmessagewithprivkey.py (hunk 3 above)
Comment (vasild) [reply test/functional/rpc_signmessagewithprivkey.py:32]: It skips 2 (`signmessagewithprivkey` has two required parameters). I think the intention here is to ensure the RPC returns an error if called with the wrong number of parameters.

Comment (ShubhamPalriwala) [reply test/functional/rpc_signmessagewithprivkey.py:32]: Alright! Makes sense then

**Review by theStack — COMMENTED** [review]
Concept ACK

**Review by vasild — APPROVED** [review]
ACK a3b559c970ada3c123ae24f21e45892734e7d494

In the OP: `s/rpc_signmessagewithaddress.py/wallet_signmessagewithaddress.py/`

**Review by theStack — APPROVED** [review]
Code-review ACK a3b559c970ada3c123ae24f21e45892734e7d494

(review was quite painless thanks to `--color-moved=dimmed-zebra`)
No big deal, but if you for any reason need to retouch, I'd suggest to also adapt the commit subject to the PR title -- "added test for disabled wallet" is very generic and doesn't say much.

**Pull request merged**